Osoitetyokalu.py

```diff
@@ -775,84 +775,105 @@ def set_popup_text(self, dlg, vkm_feature):
             road_address (str): Road address that consists of road/roadway/(road)part/distance.
         """
 
-        try:
-            #getting road coordinates and road address that are nearest to the mouse click
+        if 'tie' in vkm_feature['properties']:
+            try:
```

Comment:
Tämän koko try-except blockin voisi poistaa ja korvata esim näin:
```
if "katunimi" in  vkm_feature['properties'] and "katunumero" in vkm_feature['properties']:
    dlg.KuntanimilineEdit.setText(vkm_feature['properties']['kuntanimi'])
    dlg.KatunimilineEdit.setText((vkm_feature['properties']['katunimi'])
```
Jonka jälkeen kaikki muut sijoitukset. Myös arvojen tallentaminen ensin omaan muuttujaan on melko turhaa. Vaan ne voisi sijoittaa suoraan lineEditteihin, kuten edellä olevassa esimerkissä

Reply:
Itse asiassa koko funktion if-else blockin voisi poistaa ja lisätä if tarkastelut vain mahdollisesti puuttuville attribuuteille. Tällöin metodin koodi lyhenisi huomattavasti.